shell/platform/fuchsia/flutter/fuchsia_external_view_embedder.cc

```diff
@@ -462,58 +506,70 @@ void FuchsiaExternalViewEmbedder::SubmitFrame(
   {
     TRACE_EVENT0("flutter", "PresentSurfaces");
 
-    surface_producer_.OnSurfacesPresented(std::move(frame_surfaces));
+    surface_producer_->OnSurfacesPresented(std::move(frame_surfaces));
   }
 
   // Submit the underlying render-backend-specific frame for processing.
   frame->Submit();
 }
 
-void FuchsiaExternalViewEmbedder::EnableWireframe(bool enable) {
+void FuchsiaNextExternalViewEmbedder::EnableWireframe(bool enable) {
   session_.get()->Enqueue(
       scenic::NewSetEnableDebugViewBoundsCmd(root_view_.id(), enable));
   session_.Present();
 }
 
-void FuchsiaExternalViewEmbedder::CreateView(int64_t view_id) {
-  FML_DCHECK(scenic_views_.find(view_id) == scenic_views_.end());
-
-  ScenicView new_view = {
-      .opacity_node = scenic::OpacityNodeHACK(session_.get()),
-      .entity_node = scenic::EntityNode(session_.get()),
-      .view_holder = scenic::ViewHolder(
-          session_.get(),
-          scenic::ToViewHolderToken(zx::eventpair((zx_handle_t)view_id)),
-          "Flutter::PlatformView"),
-  };
-
-  new_view.opacity_node.SetLabel("flutter::PlatformView::OpacityMutator");
-  new_view.entity_node.SetLabel("flutter::PlatformView::TransformMutator");
-  new_view.opacity_node.AddChild(new_view.entity_node);
-  new_view.entity_node.Attach(new_view.view_holder);
-  new_view.entity_node.SetTranslation(0.f, 0.f,
-                                      -kScenicZElevationBetweenLayers);
-
-  scenic_views_.emplace(std::make_pair(view_id, std::move(new_view)));
-}
+void FuchsiaNextExternalViewEmbedder::CreateView(
+    int64_t view_id,
+    std::optional<ViewProperties> properties) {
+  raster_task_runner_->PostTask([this, view_id, properties]() {
+    FML_DCHECK(scenic_views_.find(view_id) == scenic_views_.end());
+
+    ScenicView new_view = {
+        .opacity_node = scenic::OpacityNodeHACK(session_.get()),
+        .entity_node = scenic::EntityNode(session_.get()),
+        .view_holder = scenic::ViewHolder(
+            session_.get(),
+            scenic::ToViewHolderToken(zx::eventpair((zx_handle_t)view_id)),
+            "Flutter::PlatformView"),
+    };
+
+    new_view.opacity_node.SetLabel("flutter::PlatformView::OpacityMutator");
+    new_view.entity_node.SetLabel("flutter::PlatformView::TransformMutator");
+    new_view.opacity_node.AddChild(new_view.entity_node);
+    new_view.entity_node.Attach(new_view.view_holder);
+    new_view.entity_node.SetTranslation(0.f, 0.f,
+                                        -kScenicZElevationBetweenLayers);
+    if (properties.has_value()) {
+      new_view.pending_hit_testable = properties->hit_testable;
+      new_view.pending_focusable = properties->focusable;
+    }
 
-void FuchsiaExternalViewEmbedder::DestroyView(int64_t view_id) {
-  size_t erased = scenic_views_.erase(view_id);
-  FML_DCHECK(erased == 1);
+    scenic_views_.emplace(std::make_pair(view_id, std::move(new_view)));
```

Comment:
Just curious if
scenic_views_[view_id] = std::move(new_view);
would work?

Reply:
Nice I think that would, I am just being old-fashioned